torq.q

```diff
@@ -565,6 +562,18 @@ sys:{[cmd]
 	.lg.o[`system;"executing system command: ",cmd];
 	catcherror:{[cmd;error] .lg.e[`system;"failed to execute ",cmd,": ",error];'error};
 	@[{result:system x;.lg.o[`system;"successfully executed"];result};cmd;catcherror[cmd;]]
+	/-@[{result:system x;.lg.o[`system;"successfully executed"];result};cmd;{.lg.e[`system;"failed to execute ",cmd,": ",x];'x}]
+	};
+
+// Load additional config files, check if the specific configuration environment variable exists and load the settings if it does
+loadaddcnfg:{[envvar;cnfgpath]
+	evstring:string envvar;
+	$[""~getenv[envvar];
+	.lg.o[`fileload;"environment variable ",evstring," not set, not loading app specific config"];
+	(set[cnfgpath;getenv[envvar],"/settings/"];
+	.lg.o[`fileload;"environment variable ",evstring," set, loading app specific config"];
+	.proc.loadconfig[getenv[envvar],"/settings/";] each `default,.proc.parentproctype,.proc.proctype,.proc.procname)
```

Comment:
I mentioned above switching to an `if` instead of `$` so this code doesn't need to be wrapped in a block. However atm you don't actually even have it in a block (i.e. that'd be wrapped with `[]`), instead you have it as a list (wrapped with `()`).

Key difference in this context is that a block (`[]`) will execute expressions left-to-right, but a list (`()`) will excutre right-to-left, which may cause behaviour you're not expecting:

```q
q)[-1"first";-1"second"];
first
second
q)(-1"first";-1"second");
second
first
```

(Of course, if we switch to the `if` with the early return, this is all academic anyway)

Reply:
I did test with [] aswell and still felt the functionailty was acting strangely (can't remember exactly why now, I think I was hoping to use the `cnfgpath` variable within `.proc.loadconfig`  but hopefully removing this section from any conditionals or `if` blocks etc will remove this issue) I reverted to a list purely cause it was what was there before but thanks for the clarification, and yeh won't matter changing to an `if` block anyway. Would you say in general it is better practice to use [] purely for readability then?

Reply:
the comment below is about similar stuff if you want to reply to just one

Reply:
Yeah generally best to use `[]` to combine multiple expressions within one branch of a `$` conditional (although my preference is to avoid having multiple expressions within one branch of a `$` as much as possible, as it can quickly get quite hard to follow with so many `[]` around - there are often ways to avoid it, such as using `if`s with early returns, or using function calls in the `$` branches instead)